tests/OpenAI.php

```diff
@@ -13,3 +16,17 @@
 
     expect($openAI)->toBeInstanceOf(Client::class);
 });
+
+it('accepts a custom http client', function () {
+    $client = new class implements ClientInterface
+    {
+        public function sendRequest(RequestInterface $request): ResponseInterface
+        {
+            throw new \LogicException('Not implemented');
+        }
+    };
+
+    $openAI = OpenAI::client('foo', client: $client);
```

Comment:
I feel like the test should either ensure that the passed client is actually used when consuming the API, or that putting `client: null` would provide the same "safety" here